### PR TITLE
bump edition to 2021

### PR DESCRIPTION
**Pull request by cataggar**
The edition was stable in https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html last October.

### CODE REVIEW

File: sdk/data_cosmos/src/operations/list_attachments.rs [hunk 1]
```diff
@@ -46,6 +46,7 @@ impl ListAttachmentsBuilder {
             let this = self.clone();
             let ctx = self.context.clone();
             async move {
+                let _ = &this;
```
Comment (cataggar) [inline sdk/data_cosmos/src/operations/list_attachments.rs:49]: `cargo fix --edition` added this for the 2021 edition. Why the change? What is this doing? Can it be simplified?

Comment (cataggar) [reply sdk/data_cosmos/src/operations/list_attachments.rs:49]: May be it is not needed.

**Review by rylev — APPROVED** [review]
